File: mambo-shared/mambo-protocol/mambo-protocol-client/src/main/java/org/mambo/protocol/client/types/SkillActionDescriptionCraftExtended.java


// Generated on 11/11/2012 20:41:41
package org.mambo.protocol.client.types;

import org.mambo.core.io.*;

public class SkillActionDescriptionCraftExtended extends SkillActionDescriptionCraft {
    public static final short TYPE_ID = 104;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public byte thresholdSlots;
    public byte optimumProbability;
    
    public SkillActionDescriptionCraftExtended() { }
    
    public SkillActionDescriptionCraftExtended(short skillId, byte maxSlots, byte probability, byte thresholdSlots, byte optimumProbability) {
        super(skillId, maxSlots, probability);
        this.thresholdSlots = thresholdSlots;
        this.optimumProbability = optimumProbability;
    }
    
    @Override
    public void serialize(DataWriterInterface writer) {
        super.serialize(writer);
        writer.writeByte(thresholdSlots);
        writer.writeByte(optimumProbability);
    }
    
    @Override
    public void deserialize(DataReaderInterface reader) {
        super.deserialize(reader);
        thresholdSlots = reader.readByte();
        if (thresholdSlots < 0)
            throw new RuntimeException("Forbidden value on thresholdSlots = " + thresholdSlots + ", it doesn't respect the following condition : thresholdSlots < 0");
        optimumProbability = reader.readByte();
        if (optimumProbability < 0)
            throw new RuntimeException("Forbidden value on optimumProbability = " + optimumProbability + ", it doesn't respect the following condition : optimumProbability < 0");
    }
    
}

File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/interactive/skill/SkillActionDescriptionCraftExtended.scala





















// Generated on 05/03/2013 10:35:07
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait SkillActionDescriptionCraftExtended extends SkillActionDescriptionCraft {
  override val typeId = 104

  protected trait Type extends SkillActionDescriptionCraft.Type with Serializable {
    override val typeId = SkillActionDescriptionCraftExtended.typeId

    val thresholdSlots: Byte
    val optimumProbability: Byte

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeByte(thresholdSlots)
      buf.writeByte(optimumProbability)
    }

  }

  protected def create(skillId: Short, maxSlots: Byte, probability: Byte, thresholdSlots: Byte, optimumProbability: Byte): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val skillId = buf.readShort()
    val maxSlots = buf.readByte()
    val probability = buf.readByte()
    val thresholdSlots = buf.readByte()
    val optimumProbability = buf.readByte()

    create(skillId, maxSlots, probability, thresholdSlots, optimumProbability)
  }

}

object SkillActionDescriptionCraftExtended extends SkillActionDescriptionCraftExtended {
  case class TypeImpl(skillId: Short, maxSlots: Byte, probability: Byte, thresholdSlots: Byte, optimumProbability: Byte) extends Type
  override def create(skillId: Short, maxSlots: Byte, probability: Byte, thresholdSlots: Byte, optimumProbability: Byte) = TypeImpl(skillId, maxSlots, probability, thresholdSlots, optimumProbability)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(skillId: Short, maxSlots: Byte, probability: Byte, thresholdSlots: Byte, optimumProbability: Byte) = create(skillId, maxSlots, probability, thresholdSlots, optimumProbability)

  type Target = TypeImpl

}

